### ISSUE TITLE
Rollback latest library update restrictions

### ISSUE THREAD

**Issue by makemefeelgr8** [tachiyomiorg/tachiyomi#6649]
### Describe your suggested feature

So, you were like "let us break library updates by default". And now I've missed a ton of new chapters cause of you. Why would you do this to me? Who thought it was a good idea to make library not update properly? Imagine if tachiyomi was a messaging app and you had to open every conversation and pull down to find out if there are new messages. Sounds weird.
Even more, one does not explain how to turn off this new "feature", and it's enabled by default.

https://github.com/tachiyomiorg/tachiyomi/issues/6599
Oh, and you made sure to block comments in that issue. Come on, you're not that ignorant, are you? To break library updates intentionally and then to be like "shut up and deal with it". Don't do that, please.

So, I had to dig into settings and twick some stuff here and there to make it work again. Oh, and the translation is completely wrong, it says like "choose what you would like to update", and if you choose nothing, it starts working properly.

This is the reasoning behind it:
`These changes are meant to help alleviate unnecessary traffic and ensure that sources stay available`

So, let me post my suggestions.
1. Allow comments in the said issue. Do not force your users to shut up. Making people angry is never a good idea. Furthermore, angry programmers might do a lot of damage.
2. Do not enable stuff like this silently. Add a popup that would explain the details and ask "do you want it enabled", if it's *that* critical. Be more open about it!
3. Do not enable it for manual refresh. If a user is updating a library manually, they want to see all the newest manga chapters *right now*, not in 24 hours.

Thanks for your hard work!

### Other details

Jusr a side note, totally unrelated. I've made my own fork to test some stuff. Theres a branch called feature/speedup-search, so feel free to try it out. I've changed the id to make it installable side by side. It really does wonders, searching over 200 sources ends in like 1 minute or less.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by arkon** [comment]
> Why would you do this to me?

This is already explained in #6599.

> Imagine if tachiyomi was a messaging app and you had to open every conversation and pull down to find out if there are new messages. 

This is a contrived example. The architectures are not the same, and the option is still readily available.

> Even more, one does not explain how to turn off this new "feature", and it's enabled by default.

It's mentioned in both #6599 and the [release notes](https://github.com/tachiyomiorg/tachiyomi/releases/tag/v0.13.1) that this is configurable behaviour.

> Come on, you're not that ignorant, are you? To break library updates intentionally and then to be like "shut up and deal with it". Don't do that, please.

It's also expected that people will discuss this on Discord and other issues if they have particular feedback, such as this issue.

> Oh, and the translation is completely wrong, it says like "choose what you would like to update", and if you choose nothing, it starts working properly.

We can reword it to make it clearer if you have actual, concrete suggestions.

> Allow comments in the said issue. Do not force your users to shut up. 

See earlier point.

> Furthermore, angry programmers might do a lot of damage.

Threats are not appreciated.

> Do not enable stuff like this silently.

It's in the release notes. It's perhaps unsurprising that people using a reading app don't read it. We'll try to better address it by making it more obvious what's happening rather than just showing the lowered number (if items are filtered) in the update notification.

> Do not enable it for manual refresh.

There's no actual difference in the underlying mechanism. A scheduled update is just a scheduled version of a manual refresh.

> Theres a branch called feature/speedup-search

PRs are welcome!

**Comment by Soitora** [comment]
> And now I've missed a ton of new chapters cause of you. Why would you do this to me? 

Oh no... anyways

> Who thought it was a good idea to make library not update properly?

It is updating properly, by not updating stuff you're not caught up-to-date on. It is imperfect but with our size of userbase it's a necessary thing until we can figure out a better way to handle it.

It is not a permanent and completely figured out method by any means, but we have the responsibility to do our best to alleviate for the sources used. Not legally; morally. (also helps us not to get blocked)

> Even more, one does not explain how to turn off this new "feature", and it's enabled by default.

Literally explore the settings for 2 seconds

> Oh, and the translation is completely wrong, it says like "choose what you would like to update", and if you choose nothing, it starts working properly.

The actual wording used it "Library update restrictions", and the subtext is "Only update:" and whatever items you have chosen.
If the translations are wrong, feel free to go to our Weblate and fix it.

For the rest of the comments, arkon has that covered.

**Comment by makemefeelgr8** [comment]
Yall have already canceled a similar pr once, I'm not making it again.
It's kinda like PR 6304 one, but with no restrictions at all. I kinda make it as a joke and thought it's gonna lag or crash. But somehow it did not. I guess new versions of android are good enough at handling threads and choosing max degree of concurrency. It does wonders on android 11 and 12.

**Comment by Soitora** [comment]
> It's kinda like PR 6304 one, but with no restrictions at all. 

What made you think this was a good idea? With well over a million active users this will not work
Sure 5 might be a little low but it's also balanced, but completely removing restrictions is bananas 🍌

**Comment by CrepeTF** [comment]
Awfully convenient of you to ignore the DDoS concerns

**Comment by makemefeelgr8** [comment]
> Threats are not appreciated.
Did it sound like it? Sorry, English is not my native language. I'm just watching a video about the reasoning behind PS3 getting hacked. Basically, Sony really pissed off a lot of devs and the device got hacked in no time. And now I see a pinned issue and disabled comments, and lots of people not getting to read theirs favourite manga on time. Thought it's kinda similar. I'm no programmer anyway.
> With well over a million active users this will not work
> DDoS concerns
Could you please provide some details? I mean, there are 10 sites and 10 extensions. So, what's the difference between opening all 10 sites at once and opening em 1 by 1? It's not like 10 extensions are pointing to 1 site...

**Comment by makemefeelgr8** [comment]
> It is updating properly, by not updating stuff you're not caught up-to-date on

Oh, thanks for explaining. I get it now. Let me share my UX. You know, how they end manga chapters in the most interesting moment? And then I have to wait for a few days until the next one gets released. So, it's kind of disappointing to read manga as soon as 1 new chapter appears. This is why I usually wait until there are 3 or 4 new chapters, and then read them all at once.

So, I suggest to add something smarter. For example, one can consider a title is abandoned in case there are at least 10 (or over 10% from total) unread chapters.

**Comment by CarlosEsco** [comment]
> > It is updating properly, by not updating stuff you're not caught up-to-date on
> 
> Oh, thanks for explaining. I get it now. Let me share my UX. You know, how they end manga chapters in the most interesting moment? And then I have to wait for a few days until the next one gets released. So, it's kind of disappointing to read manga as soon as 1 new chapter appears. This is why I usually wait until there are 3 or 4 new chapters, and then read them all at once.
> 
> So, I suggest to add something smarter. For example, one can consider a title is abandoned in case there are at least 10 (or over 10% from total) unread chapters.

Could you open a new issue with this request and close this unnecessarily aggressive issue ?

**Comment by makemefeelgr8** [comment]
> It's in the release notes. It's perhaps unsurprising that people using a reading app don't read it.

Yeah, people rarely read these. One would usually open an app and expect to read manga, not spend 10 minutes reading a wall of text. Especially cause there's like 70% of text an average person would never understand.
Like: `Truncate MAL search queries to first 64 characters to avoid HTTP 400 errors`
Try asking your non-tech-savvy relatives what is MAL and what's HTTP 400. People have no idea.
Also, there's a thing called `breaking changes`, so you could written it with huge red letters and make it human readable.

Here's an awesome example of release notes:
https://code.visualstudio.com/updates/v1_64
- It has zero technical info.
- Every feature is described in human readable format.
- There is a little manual with screenshots accompanying every feature.

> making it more obvious what's happening rather than just showing the lowered number

Oh, I've seen that. I try to update it manually, and the notification flashes for like half a second saying 3 out of 3 updated. Instead of saying 98 out of 98. I wouldn't call it intuitive.

> We can reword it to make it clearer if you have actual, concrete suggestions

Nah, I'm not that good at English. But I find it kinda confusing, so it may be a good idea to tweak it a bit.

> A scheduled update is just a scheduled version of a manual refresh.

So, I guess it makes sense to implement a separate workflow for the manual update that would actually update all the categories.

> close this unnecessarily aggressive issue

I wonder what reaction and feedback did you guys expect after silently turning on this downgrade by default? Did you think that users would be like "Oh, what a great feature!", "I'm glad I had to change settings to make my library update properly", "I'm so grateful that I've missed new chapters"?
Did this feature:
- Make an app easier to use? No.
- Add something new? No.
- Improve UX? No.

Yes, I understand why you've implement it this way. Yes, there was an issue you had to fix. But this is definitely not a proper solution. I mean, you acknowledge it yourself:
> It is not a permanent and completely figured out method by any means

So, this is why I've voiced my opinion loud and clear, and created a request to roll it back. I still think it should be rolled back, that's why I'm not closing this one. It's up to you to close this issue, if you think that this feature works fine and new UX is up to your standards.

**Comment by arkon** [comment]
> Here's an awesome example of release notes:

There's plenty of content in there that would make no sense to a person who isn't in the target audience of a code editor.

I can certainly try to dumb down the release notes more, but that's best effort.

> So, I guess it makes sense to implement a separate workflow for the manual update that would actually update all the categories.

This shows that it's still unclear to most people _why_ this is being done. It avoids calls that are typically deemed unnecessary for what should be a "normal" use case. If you feel that you _need_ to check for chapters from entries that aren't ongoing or you haven't actually fully caught up on yet anyway, you can still configure it to do so. Otherwise, one-off manual refreshes work too.

> I wonder what reaction and feedback did you guys expect after silently turning on this downgrade by default?

This is all within expectations. The UX suffers a bit to help ensure sources are even available for people to use. It's not uncommon for sources to be unhappy with the amount of traffic coming from Tachiyomi and actively work on breaking the integration.

> So, this is why I've voiced my opinion loud and clear, and created a request to roll it back.

This is an understandably naive suggestion that we see from people who don't understand how the app works and the implications of their actions on the grander scale, which I explained above.

If you want to open issues with actionable improvements, please do so.
